src/process/message_processor.go

```diff
@@ -5,70 +5,97 @@ import (
 	"BGPAlert/common"
 	"BGPAlert/config"
 	"fmt"
+	"strconv"
 	"time"
 )
 
 // Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, config *config.Configuration) error {
-	maximumBuckets := config.WindowSize
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.Message, config *config.Configuration) error {
+	// Parse windowSize from config
+	var maximumBuckets int
+	// If windowSize can't be parsed to an int -> just default to 30
+	if windowSize, err := strconv.ParseInt(config.WindowSize, 10, 64); err != nil {
+		fmt.Println("Inputted config size is not a number, defaulting to 30")
```

Comment:
hmm - I think we should error the program if the valid specified is not a valid integer. We should only default to 30 if the value has been omitted from the config file.

Reply:
^editing config validation to error out;